### PR TITLE
rpc: add `BadBlock` type to `debug_getbadblocks` return type

### PR DESCRIPTION
**Pull request by EdwardJES**
This PR adds `BadBlock` as a new type and adds it to the return type to `debug_getBadBlocks`. Addresses https://github.com/alloy-rs/alloy/issues/1560

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-eth/src/block.rs [hunk 1]
```diff
@@ -105,11 +104,11 @@ pub struct Header {
     /// Timestamp
     #[cfg_attr(feature = "serde", serde(default, with = "alloy_serde::quantity"))]
     pub timestamp: u64,
+    /// Extra data
```
Comment (EdwardJES) [inline crates/rpc-types-eth/src/block.rs:107]: Reordered to allow deriving of `Rlp` traits.

File: crates/rpc-types-eth/src/block.rs [hunk 2]
```diff
@@ -70,9 +68,10 @@ impl<T> Block<T> {
 
 /// Block header representation.
 #[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
-#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash, RlpDecodable, RlpEncodable)]
```
Comment (EdwardJES) [inline crates/rpc-types-eth/src/block.rs:71]: Will these have any conflicts with https://github.com/alloy-rs/alloy/blob/bc2040d20216a8d14ff87c31c34eff2a74635ce0/crates/consensus/src/header.rs#L351?

File: crates/rpc-types-eth/src/block.rs [hunk 3]
```diff
@@ -462,6 +461,19 @@ impl<T: TransactionResponse, H: HeaderResponse> BlockResponse for Block<T, H> {
     }
 }
 
+/// Bad block representation.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
+pub struct BadBlock {
```
Comment (EdwardJES) [inline crates/rpc-types-eth/src/block.rs:468]: Unsure if this deserved it's own file or if this is the most suitable location.

File: crates/rpc-types-eth/src/block.rs (hunk 2 above)
Comment (mattsse) [inline crates/rpc-types-eth/src/block.rs:71]: no rlp for rpc types

**Review by mattsse — CHANGES_REQUESTED** [review]
this pr should be limited to the new badblock type and not change any existing types.

File: crates/rpc-types-eth/src/block.rs (hunk 2 above)
Comment (EdwardJES) [reply crates/rpc-types-eth/src/block.rs:71]: Cool, got it. Will change. 

What would be the best way to calculate the rlp encoded header for the `rlp` field? 

Ty 🙏

Comment (EdwardJES) [reply crates/rpc-types-eth/src/block.rs]: Good point, my mistake. Fixed.

**Comment by EdwardJES** [comment]
Thank you, @mattsse. Have addressed the comments.

**Review by mattsse — APPROVED** [review]

**Comment by EdwardJES** [comment]
Fixing clippy now.

**Pull request merged**